### DISCUSSION TITLE
Model hooks requiring additional data

### DISCUSSION THREAD

**Discussion by rtdavis22** [jeremyevans/sequel#2032] [General]
Hello, I have a Sequel Model that references files stored in a filesystem and a `before_destroy` hook that deletes the files. It looks something like this:
```
class File < Sequel::Model
  model_schema do
    primary_key :id
    String :path
  end

  def before_destroy
    FileStore.delete_file(path)
    super
  end
end
```

This works alright, but the FileStore API was updated to also require a context object to be passed in. Any advice on what I should do here?

**Comment by jeremyevans** [comment]
Assuming that the context is something external to the model class, easiest thing would be to add an attr_accessor or something that you can use to set the context, and and call the reader method inside before_destroy.  Then call the writer method before calling destroy on the model instance.
